packages/dev/sharedUiComponents/src/lines/optionsLineComponent.tsx

```diff
@@ -80,44 +83,103 @@ export class OptionsLineComponent extends React.Component<IOptionsLineComponentP
     }
 
     updateValue(valueString: string) {
-        const value = this.props.valuesAreStrings ? valueString : parseInt(valueString);
+        let value = this.props.valuesAreStrings ? valueString : parseInt(valueString);
+
+        if (isNaN(Number(value))) {
+            for (let i = 0; i < this.props.options.length; i++) {
+                if (this.props.options.at(i)?.label === valueString) {
+                    value = Number(this.props.options.at(i)?.value);
+                }
+            }
+        }
+
+        this.forceUpdate();
+        if (value === 0) {
+            this.setState({ addCustom: true });
+        }
+
         this._localChange = true;
 
         const store = this.props.extractValue ? this.props.extractValue(this.props.target) : this.props.target[this.props.propertyName];
 
         if (!this.props.noDirectUpdate) {
             this.props.target[this.props.propertyName] = this._remapValueOut(value as number);
         }
-        this.setState({ value: value });
-
-        if (this.props.onSelect) {
-            this.props.onSelect(value);
+        if (value != 0) {
+            this.setState({ value: value });
+            if (this.props.onSelect) {
+                this.props.onSelect(value);
+            }
+        } else {
+            if (this.props.onSelect) {
+                this.props.onSelect(this.state.value);
+            }
         }
 
         const newValue = this.props.extractValue ? this.props.extractValue(this.props.target) : this.props.target[this.props.propertyName];
 
         this.raiseOnPropertyChanged(newValue, store);
     }
 
+    updateCustomValue() {
+        this.setState({ addCustom: false });
+    }
+
     render() {
-        return (
-            <div className={"listLine" + (this.props.className ? " " + this.props.className : "")}>
-                {this.props.icon && <img src={this.props.icon} title={this.props.iconLabel} alt={this.props.iconLabel} color="black" className="icon" />}
-                <div className="label" title={this.props.label}>
-                    {this.props.label}
+        if (this.state.addCustom) {
+            return (
+                <div className={"listLine" + (this.props.className ? " " + this.props.className : "")}>
+                    {this.props.icon && <img src={this.props.icon} title={this.props.iconLabel} alt={this.props.iconLabel} color="black" className="icon" />}
+                    <div className="label" title={this.props.label}>
+                        {this.props.label}
+                    </div>
+                    <div className="options">
+                        <input
+                            type="text"
+                            placeholder="Enter a custom font here"
+                            id="customFont"
+                            onKeyDown={(event) => {
+                                event.keyCode === 13 && this.props.addVal != undefined

```

Comment:
Keycode is deprecated: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode, it's recommended to use code instead: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code

But imho I prefer key https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key as its easier to read (it's easier to check key === "A" than remember the key code for A is 65 )